graveler/graveler.go

```diff
@@ -1397,6 +1427,119 @@ func (g *Graveler) getCommitsForMerge(ctx context.Context, repositoryID Reposito
 	return fromCommit, toCommit, baseCommit, nil
 }
 
+func (g *Graveler) LoadCommits(ctx context.Context, repositoryID RepositoryID, metaRangeID MetaRangeID) error {
+	repo, err := g.GetRepository(ctx, repositoryID)
+	if err != nil {
+		return err
+	}
+	iter, err := g.CommittedManager.List(ctx, repo.StorageNamespace, metaRangeID)
+	if err != nil {
+		return err
+	}
+	defer iter.Close()
+	for iter.Next() {
+		rawValue := iter.Value()
+		if rawValue == nil {
+			break // better check for error
+		}
+		commit := &CommitData{}
+		err := proto.Unmarshal(rawValue.Data, commit)
+		if err != nil {
+			return err
+		}
+		parents := make(CommitParents, len(commit.GetParents()))
+		for i, p := range commit.GetParents() {
+			parents[i] = CommitID(p)
+		}
+		commitID, err := g.RefManager.AddCommit(ctx, repositoryID, Commit{
+			Committer:    commit.GetCommitter(),
+			Message:      commit.GetMessage(),
+			MetaRangeID:  MetaRangeID(commit.GetMetaRangeId()),
+			CreationDate: commit.GetCreationDate().AsTime(),
+			Parents:      parents,
+			Metadata:     commit.GetMetadata(),
+		})
+		if err != nil {
+			return err
+		}
+		// integrity check that we get for free!
+		if commitID != CommitID(commit.Id) {
+			return fmt.Errorf("commit ID does not match for %s: %w", commitID, ErrInvalidCommitID)
+		}
+	}
+	if iter.Err() != nil {
+		return iter.Err()
+	}
+	return nil
+}
+
+func (g *Graveler) LoadBranches(ctx context.Context, repositoryID RepositoryID, metaRangeID MetaRangeID) error {
+	repo, err := g.GetRepository(ctx, repositoryID)
+	if err != nil {
+		return err
+	}
+	iter, err := g.CommittedManager.List(ctx, repo.StorageNamespace, metaRangeID)
+	if err != nil {
+		return err
+	}
+	defer iter.Close()
+	for iter.Next() {
+		rawValue := iter.Value()
+		if rawValue == nil {
+			break // better check for error
```

Comment:
Not sure whether this is a TODO or an instruction; consider making it a clearer imperative, e.g. (if the comment I wrote is true...)
```suggestion
			break // report deserialization error
```

But then... what if the iterator returns `nil` but no error?  Need to test that this iterator cannot do this, otherwise we might successfully "import" a repo incompletely.